### PR TITLE
Added the license checker (part of #2430)

### PR DESCRIPTION
**Pull request by kirovboris**
/cc @AndreyBelym @VasilyStrelyaev

### CODE REVIEW

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Comment by kirovboris** [comment]
ping @VasilyStrelyaev, check the strings please

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

File: test/dependency-licenses-checker.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const INCOMPATIBLE_LICENSES_RE = /GPL/i;
+
+const hasIncompatibleLicenses = license => INCOMPATIBLE_LICENSES_RE.test(license);
+const hasUnknownLicenses      = license => PERMISSIVE_LICENSES.indexOf(license) === -1;
+
+const modulePath = path.join(__dirname, '../');
+
+function checkLicense (license, module) {
+    const unknownModuleAssertionMsg       = `The module '${module}' contains an unknown license '${license}'. Update the permissive licenses list only if it's a known permissive license`;
+    const incompatibleLicenseAssertionMsg = `The module '${module}' contains an incompatible license: '${license}' and can't be used with testcafe !!!`;
```
Comment (VasilyStrelyaev) [inline test/dependency-licenses-checker.js]:
```
    const incompatibleLicenseAssertionMsg = `Alert! The module '${module}' contains an incompatible license '${license}' and cannot be used with TestCafe!`;
```

File: test/dependency-licenses-checker.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+const INCOMPATIBLE_LICENSES_RE = /GPL/i;
+
+const hasIncompatibleLicenses = license => INCOMPATIBLE_LICENSES_RE.test(license);
+const hasUnknownLicenses      = license => PERMISSIVE_LICENSES.indexOf(license) === -1;
+
+const modulePath = path.join(__dirname, '../');
+
+function checkLicense (license, module) {
+    const unknownModuleAssertionMsg       = `The module '${module}' contains an unknown license '${license}'. Update the permissive licenses list only if it's a known permissive license`;
```
Comment (VasilyStrelyaev) [inline test/dependency-licenses-checker.js]:
```
    const unknownModuleAssertionMsg       = `The module '${module}' contains an unknown license '${license}'. You can add this license to the list of known permissive licenses. But be cautious and do this only if you are sure that this license is compatible with TestCafe`;
```

**Comment by kirovboris** [comment]
FPR

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Pull request merged**